### PR TITLE
More Heap unit tests:

### PR DESCRIPTION
**Pull request by grandizzy**
- rename methods to upsert and remove in Heap test contract
- add tests for upsert (+ fuzzy)
- add require check tests for inserting or updating TP to 0

### CODE REVIEW

File: src/_test/Heap.t.sol [hunk 1]
```diff
@@ -94,96 +94,103 @@ contract HeapTest is DSTestPlus {
...
-        assertEq(_pool.getMaxTp(), 0);
+        vm.expectRevert("H:I:VAL_EQ_0");
+        _pool.upsertTp(b1, 0);
     }
 
-    function testHeapUpdateTps() public {
-        // TODO
+    function testHeapUpsertFuzzy(uint256 tp_, address borrower_) public {
+        tp_ = bound(tp_, 1, 1e18);
+        _pool.upsertTp(borrower_, tp_);
```
Comment (MikeHathaway) [inline src/_test/Heap.t.sol]: Might be interesting to capture current `_pool.getMaxTp()`, check if fuzzed tp is > than, and see if the max changes post upsert as part of this fuzzy test

**Review by MikeHathaway — APPROVED** [review]
LGTM

File: src/_test/Heap.t.sol (hunk 1 above)
Comment (grandizzy) [reply src/_test/Heap.t.sol]: yep, good point, just realized that this fuzzy test is not relevant as it only inserts one random node in an empty heap, will rework

Comment (grandizzy) [reply src/_test/Heap.t.sol]: I removed fuzzy test with https://github.com/ajna-finance/contracts/pull/238/commits/c4d34d9408769e3c61d84ee0be031834d96fdb5f proper PR pending

**Review by EdNoepel — APPROVED** [review]
LGTM

**Pull request merged**